src/program.ts

```diff
@@ -0,0 +1,286 @@
+import {
+  Keypair,
+  Connection,
+  BpfLoader,
+  BPF_LOADER_PROGRAM_ID,
+  SystemProgram,
+  Transaction,
+  sendAndConfirmTransaction,
+  TransactionInstruction,
+} from '@solana/web3.js';
+import { ethers } from 'ethers';
+import { ContractDeployResult } from 'src';
+
+import { Contract, ContractDeployOptions } from './contract';
+import {
+  LogsParser,
+  LogCallback,
+  EventCallback,
+  parseTxError,
+  parseTxLogs,
+} from './logs';
+
+export class Program {
+  private logs: LogsParser;
```

Comment:
I think we should prefer `public` or `protected` over `private` in general unless there's a clear reason to encapsulate. This code provides a base layer, so it should be easy to extend.